archunit/src/test/java/com/tngtech/archunit/core/importer/ClassFileImporterGenericFieldTypesTest.java

```diff
@@ -0,0 +1,522 @@
+package com.tngtech.archunit.core.importer;
+
+import java.io.File;
+import java.io.Serializable;
+import java.lang.ref.Reference;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.tngtech.archunit.core.domain.JavaClasses;
+import com.tngtech.archunit.core.domain.JavaType;
+import com.tngtech.archunit.testutil.ArchConfigurationRule;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static com.tngtech.archunit.testutil.Assertions.assertThatType;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteClass.concreteClass;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteParameterizedType.parameterizedType;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteTypeVariable.typeVariable;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteTypeVariableArray.typeVariableArray;
+import static com.tngtech.archunit.testutil.assertion.ExpectedConcreteType.ExpectedConcreteWildcardType.wildcardType;
+
+@RunWith(DataProviderRunner.class)
+public class ClassFileImporterGenericFieldTypesTest {
+
+    @Rule
+    public final ArchConfigurationRule configurationRule = new ArchConfigurationRule().resolveAdditionalDependenciesFromClassPath(false);
+
+    @Test
+    public void imports_non_generic_field_type() {
+        class NonGenericFieldType {
+        }
+        @SuppressWarnings("unused")
+        class SomeClass {
+            NonGenericFieldType field;
+        }
+
+        JavaType fieldType = new ClassFileImporter().importClass(SomeClass.class).getField("field").getType();
+
+        assertThatType(fieldType).as("generic field type").matches(NonGenericFieldType.class);
+    }
+
+    @Test
+    public void imports_generic_field_type_with_one_type_argument() {
+        @SuppressWarnings("unused")
```

Comment:
The `@SuppressWarnings` here simply covers the case of the unused type parameter. That's why `NonGenericFieldType` doesn't need it. From my point of view within ArchUnit I would generally suppress all warnings of local classes of `@Test` methods :joy: 
But what I think makes sense is to not have any warnings in valid code :man_shrugging: If those warnings don't make sense for our case, because the info is dug up from the byte code and thus still valid, we should suppress them, no?